partiql-planner/src/main/kotlin/org/partiql/planner/typer/PlanTyper.kt

```diff
@@ -759,6 +776,63 @@ internal class PlanTyper(
                 false -> StaticType.ANY
             }
         }
+
+        /**
+         * Resolution and typing of aggregation function calls.
+         *
+         * I've chosen to place this in RexTyper because all arguments will be typed using the same locals.
+         * There's no need to create new RexTyper instances for each argument. There is no reason to limit aggregations
+         * to a single argument (covar, corr, pct, etc.) but in practice we typically only have single <value expression>.
+         *
+         * This method is _very_ similar to scalar function resolution, so it is temping to DRY these two out; but the
+         * separation is cleaner as the typing of NULLS is subtly different.
+         *
+         * SQL-99 6.16 General Rules on <set function specification>
+         *     Let TX be the single-column table that is the result of applying the <value expression>
+         *     to each row of T and eliminating null values <--- all NULL values are eliminated as inputs
+         */
+        public fun resolveAgg(agg: Agg.Unresolved, arguments: List<Rex>): Pair<Rel.Op.Aggregate.Call, StaticType> {
+            var missingArg = false
+            val args = arguments.map {
+                val arg = visitRex(it, null)
+                if (arg.type == MissingType) missingArg = true
+                arg
+            }
+
+            //
+            if (missingArg) {
+                handleAlwaysMissing()
+                return relOpAggregateCall(agg, listOf(rexErr("MISSING"))) to MissingType
+            }
+
+            // Try to match the arguments to functions defined in the catalog
+            return when (val match = env.resolveAgg(agg, args)) {
+                is FnMatch.Ok -> {
+                    // Found a match!
+                    val newAgg = aggResolved(match.signature)
+                    val newArgs = rewriteFnArgs(match.mapping, args)
+                    val returns = newAgg.signature.returns
+
+                    // Return type with calculated nullability
+                    var type = when {
+                        newAgg.signature.isNullable -> returns.toStaticType()
+                        else -> returns.toNonNullStaticType()
+                    }
+
+                    // Some operators can return MISSING during runtime
+                    if (match.isMissable) {
+                        type = StaticType.unionOf(type, StaticType.MISSING)
```

Comment:
While it's not an explicit requirement within PlanTyper, several functions in the PlanTyper flatten their union types before returning. Helps avoid accidental breaks in the future.

Here, you could potentially have a `type = union(union(type, NULL), MISSING)` -- which, I don't think all functions account for. When you call `allTypes`, you typically wouldn't expect any `AnyOfType`'s to be part of the return.

Reply:
That's very odd StaticType behavior. Union is an associative operation you'd expect

```
(A U B) U (B U C)    <-->  A U B U C
``` 

The need to flatten is a workaround for StaticType union not being associative